airflow/providers/google/cloud/operators/bigquery.py

```diff
@@ -1181,7 +1191,13 @@ def execute(self, context: Context):
             ]
         else:
             raise AirflowException(f"argument 'sql' of type {type(str)} is neither a string nor an iterable")
-        context["task_instance"].xcom_push(key="job_id", value=job_id)
+        job_id_params = {
+            "job_id": job_id,
+            "project_id": self.hook.project_id,
+            "location": self.location if self.location else "US",
+        }
+        context["task_instance"].xcom_push(key="job_id", value=job_id_params)
```

Comment:
The key name is a bit misleading, as it now includes more information - maybe rename it to `job_id_params` as well?

Reply:
changed